### ISSUE TITLE
Integration with mysql is throwing exception

### ISSUE THREAD

**Issue by sanjibislam** [cloudfoundry/uaa#284]
When I try to integrate with mysql, I get exception error. Please take a look at the attached log and uaa.yaml files.
[uaa.log.txt](https://github.com/cloudfoundry/uaa/files/76105/uaa.log.txt)
[uaa.yml.txt](https://github.com/cloudfoundry/uaa/files/76106/uaa.yml.txt)

Below is the root cause from http://localhost:8080/uaa/:-
org.springframework.beans.factory.BeanDefinitionStoreException: Invalid bean definition with name 'org.cloudfoundry.identity.uaa.config.YamlConfigurationValidator#0' defined in ServletContext resource [/WEB-INF/spring-servlet.xml]: Could not resolve placeholder 'environmentYamlKey' in string value "${environmentYamlKey}"; nested exception is java.lang.IllegalArgumentException: Could not resolve placeholder 'environmentYamlKey' in string value "${environmentYamlKey}"

**Comment by fhanik** [comment]
How are you starting up the UAA? Looks like it is misconfigured, and we need to figure out why.

    ./gradlew run

Using Java 8 is all you need to start it up properly.

**Comment by fhanik** [comment]
And please post any additional configuration you may have modified. It could be a syntactical error there.

**Comment by sanjibislam** [comment]
I had downloaded the war file (cloudfoundry-identity-uaa-2.7.1.war), renamed as uaa.war and deployed to my local tomcat (meaning, copied uaa.war to tomcat's webapps folder and restarted tomcat).
The environment variable UAA_CONFIG_PATH was set to the folder that contains uaa.yml (attached with my original issue post).

I had also tried to start uaa from source code (uaa-master) using gradle (as you mentioned ./gradlew run) but I saw the same exception.

**Comment by sanjibislam** [comment]
BTW, my environment is running on windows 10 OS

**Comment by sanjibislam** [comment]
@fhanik Is there any update to resolve this issue?

**Comment by fhanik** [comment]
#mysql commands that were run
    create database uaa;

I don't think you intended to uncomment the second line, that line breaks your configuration

    #mysql commands that were run
    #create database uaa;

**Comment by sanjibislam** [comment]
Thanks for the clue. I expected when uaa is launched by the app server (ex. tomcat), the database (uaa) will be created if it does not exist in mysql server and therefore, I uncommented that line in uaa.yml
Looks like currently there is no support for that.
After I created the uaa database from mysql workbench (and left "create database" line commented), everything starts to work.
@fhanik Thanks for your help to figure this one out.
I am all set for now and this issue will be closed.
